ui/src/components/head/__inline__/loadFonts.js

```diff
@@ -0,0 +1,376 @@
+                ): void => {
+                    const xhr = new XMLHttpRequest();
+
+                    // JSONP callback
+                    window.guFont = fontData => fontData.css; // eslint-disable-line no-param-reassign
+
+                    xhr.open('GET', url, true);
+
+                    xhr.onreadystatechange = () => {
+                        if (xhr.readyState === 4 && xhr.status === 200) {
+                            const css = eval(xhr.responseText); // eslint-disable-line no-eval
+                            useFont(el, css);
+                            saveFont(fontName, fontHash, css);
+                        }
+                    };
+
+                    xhr.send();
+                };
+
+                // down to business
+                // the target for each font and holders of all the necessary metadata
+                // are some style elements in the head, all identified by a .webfont class
+                const fonts = document.getElementsByClassName('webfont');
+
+                const hinting =
+                    fontHinting === 'Off' ? '' : `hinted-${fontHinting}-`;
+
+                const urlAttribute = `data-cache-file-${hinting}${fontFormat}`;
+
+                [...fonts].forEach(font => {
+                    const fontURL = font.getAttribute(urlAttribute);
+
+                    if (fontURL) {
+                        const fontInfo = fontURL.match(
+                            /fonts\/([^/]*?)\/?([^/]*)\.(woff2|woff|ttf).json$/
+                        );
+                        if (fontInfo && fontInfo.length >= 3) {
+                            const fontName = fontInfo[2];
+                            const fontHash = fontInfo[1];
+                            const fontData = localStorage.getItem(
+                                fontStorageKey(fontName, fontHash)
+                            );
+                            if (fontData) {
+                                useFont(font, JSON.parse(fontData).value);
+                            } else {
+                                fetchFont(fontURL, font, fontName, fontHash);
+                            }
+                        }
+                    }
+                });
+
+                return true;
+            }
+        } catch (e) {
+            // do nothing
+        }
+
+        return false;
+    };
+
+    // Load fonts by injecting a `link` element.
+    const loadFontsAsynchronously = (): void => {
+        try {
+            const scripts = document.getElementsByTagName('script');
+            const thisScript = scripts[scripts.length - 1];
+            const fonts = document.createElement('link');
+
+            fonts.rel = 'stylesheet';
+            fonts.className = 'webfonts';
+
+            // show cleartype-hinted for Windows XP-7 IE, autohinted for non-IE
+            fonts.href =
+                window.guardian.config.stylesheets.fonts[
+                    `hinting${fontHinting}`
+                ].kerningOn;
```

Comment:
This `kerningOn` thingie... If it’s here to make sure our CSS is switching on kerning for all browsers that do not switch it on by default (Safari? Oldies?), then it’s needed. If it’s a leftover from the times when we had every font in no-kern flavour – it can die, as we decided not to use those at all, because the impact on design was to great.